File: code/1137.n-th-tribonacci-number.php
<?php
/*
 * @lc app=leetcode id=1137 lang=php
 *
 * [1137] N-th Tribonacci Number
 */

// @lc code=start
class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function tribonacci($n) {
        $answers = [0, 1, 1];
        for ($i = 0; $i <= $n; $i++) {
            $answers[$i + 3] = $answers[$i] + $answers[$i + 1] + $answers[$i + 2];
        }

        return $answers[$n];
    }
}
// @lc code=end

File: code/1137.n-th-tribonacci-number.go
package code

/*
 * @lc app=leetcode id=1137 lang=golang
 *
 * [1137] N-th Tribonacci Number
 */

// @lc code=start
func tribonacci(n int) int {
	if n == 0 {
		return 0
	}
	if n == 1 || n == 2 {
		return 1
	}

	a, b, c := 0, 1, 1
	for i := 3; i <= n; i++ {
		next := a + b + c
		a, b, c = b, c, next
	}
	return c
}

// @lc code=end
